givenergy_modbus_async/model/plant.py

```diff
@@ -105,12 +111,34 @@ def inverter(self) -> Inverter:
 
     @property
     def batteries(self) -> list[Battery]:
-        """Return Battery models for the Plant."""
-        return [
-            Battery.from_orm(self.register_caches[i + self.slave_address])
-            for i in range(self.number_batteries)
-        ]
+        """Return LV Battery models for the Plant."""
+        if 0x32 in self.register_caches.keys():
+            return [
+                Battery.from_orm(self.register_caches[i + 0x32])
+                for i in range(self.number_batteries)
+            ]
+        else:
+            return None
+
 
+    @property
+    def hvbatteries(self) -> list[HVBattery]:
+        """Return HV Battery models for the Plant."""
+        if 0x50 in self.register_caches.keys():
+            return [
+                HVBattery.from_orm(self.register_caches[i + 0x50])
+                for i in range(self.number_batteries)
+            ]
+        else:
+            return None
+
+    @property
+    def bcu(self) -> list[HVBCU]:
+        """Return HV Battery models for the Plant."""
+        if 0x50 in self.register_caches.keys():
```

Comment:
Should that have been  0x70 in caches.keys() ?
Also, is it sufficient to test  0x70 in self.register_caches  ?  No need to extract the keys as a list and then check that the key is in that list - you can just test for the key directly 'in' the dictionary, AFAIK